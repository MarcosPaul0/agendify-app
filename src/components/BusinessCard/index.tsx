import { TouchableOpacity, View, Text } from 'react-native';
import { Image } from 'expo-image';
import { Rating } from '@components/Rating';
import { useRouter } from 'expo-router';
import { getImageUrl } from '@utils/getImageUrl';
import { IBusinessCardProps } from './interfaces/businessCardProps.interface';

export function BusinessCard({
  imageUrl,
  title,
  description,
  favoriteIsActive,
  redirectTo,
  rating,
  totalRatings,
}: IBusinessCardProps) {
  const router = useRouter();

  // const favoriteIconStyle = isFavorite ? 'fill' : 'regular';
  const titleStyle = favoriteIsActive ? 'mr-4' : 'mr-0';

  function redirectToScreen() {
    router.push(redirectTo);
  }

  return (
    <TouchableOpacity
      onPress={redirectToScreen}
      className={`
      w-full p-3 flex-row mb-4 items-center
      bg-GRAY_100 border border-GRAY_400 rounded-xl
    `}
    >
      <Image source={getImageUrl(imageUrl)} className="w-20 h-20 rounded-lg" />

      {favoriteIsActive && (
        <View className="ml-1 absolute right-2 top-2">
          {/* <Heart weight={favoriteIconStyle} color={COLORS.ERROR} /> */}
        </View>
      )}

      <View className="pl-4 flex-1 items-start">
        <View className="border-b border-GRAY_400 pb-1 w-full">
          <Text
            className={`
              ${titleStyle} text-md font-bold text-BLUE_800
            `}
            numberOfLines={1}
          >
            {title}
          </Text>
        </View>

        <Rating
          name={title}
          rating={rating}
          numberOfRatings={totalRatings}
          showTotalRatings
        />

        <View>
          <Text className="text-xs text-GRAY_600" numberOfLines={2}>
            {description}
          </Text>
        </View>
      </View>
    </TouchableOpacity>
  );
}
