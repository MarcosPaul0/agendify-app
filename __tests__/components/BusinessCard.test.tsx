import { render } from '@testing-library/react-native';
import { BusinessCard } from '@components/BusinessCard';

jest.mock('expo-router', () => {
  const currentRouter = jest.requireActual('expo-router');

  return {
    ...currentRouter,
    useRouter: jest.fn().mockReturnValue({
      push: jest.fn().mockImplementation((path: string) => path),
    }),
  };
});

describe('Business Card', () => {
  it('Match with snapshot', async () => {
    const tree = render(
      <BusinessCard
        description="fake-description"
        imageUrl="fake-image-url"
        rating={5}
        redirectTo="fake-redirection"
        title="fake-title"
        totalRatings={5}
        isFavorite
        favoriteIsActive
      />
    );

    expect(tree).toMatchSnapshot();
  });
});
