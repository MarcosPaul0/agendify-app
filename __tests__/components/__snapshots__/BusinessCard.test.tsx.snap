// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Business Card Match with snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#F5F5F5",
      "borderBottomColor": "#CED4DA",
      "borderBottomLeftRadius": 12,
      "borderBottomRightRadius": 12,
      "borderBottomWidth": 1,
      "borderLeftColor": "#CED4DA",
      "borderLeftWidth": 1,
      "borderRightColor": "#CED4DA",
      "borderRightWidth": 1,
      "borderTopColor": "#CED4DA",
      "borderTopLeftRadius": 12,
      "borderTopRightRadius": 12,
      "borderTopWidth": 1,
      "flexDirection": "row",
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 12,
      "paddingLeft": 12,
      "paddingRight": 12,
      "paddingTop": 12,
      "width": "100%",
    }
  }
>
  <ViewManagerAdapter_ExpoImage
    borderBottomLeftRadius={8}
    borderBottomRightRadius={8}
    borderTopLeftRadius={8}
    borderTopRightRadius={8}
    height={80}
    onError={[Function]}
    onLoad={[Function]}
    onLoadStart={[Function]}
    onProgress={[Function]}
    proxiedProperties={
      {
        "contentFit": "cover",
        "contentPosition": {
          "left": "50%",
          "top": "50%",
        },
        "placeholder": [],
        "source": [
          {
            "uri": "http://10.0.2.2:3333/fake-image-url",
          },
        ],
        "tintColor": undefined,
        "transition": null,
      }
    }
    style={
      {
        "borderBottomLeftRadius": 8,
        "borderBottomRightRadius": 8,
        "borderTopLeftRadius": 8,
        "borderTopRightRadius": 8,
        "height": 80,
        "width": 80,
      }
    }
    width={80}
  />
  <View
    style={
      [
        {
          "marginLeft": 4,
        },
        {
          "position": "absolute",
        },
        {
          "right": 8,
        },
        {
          "top": 8,
        },
      ]
    }
  />
  <View
    style={
      [
        {
          "paddingLeft": 16,
        },
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "alignItems": "flex-start",
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "borderBottomWidth": 1,
          },
          {
            "borderBottomColor": "#CED4DA",
            "borderLeftColor": "#CED4DA",
            "borderRightColor": "#CED4DA",
            "borderTopColor": "#CED4DA",
          },
          {
            "paddingBottom": 4,
          },
          {
            "width": "100%",
          },
        ]
      }
    >
      <Text
        numberOfLines={1}
        style={
          [
            {
              "marginRight": 16,
            },
            {
              "fontWeight": "700",
            },
            {
              "color": "#18273C",
            },
          ]
        }
      >
        fake-title
      </Text>
    </View>
    <View
      style={
        [
          {
            "flexDirection": "row",
          },
          {
            "alignItems": "center",
          },
          {
            "paddingBottom": 4,
            "paddingTop": 4,
          },
        ]
      }
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#FFD015"
        fill="#FFD015"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#FFD015"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294955029,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#FFD015"
        fill="#FFD015"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#FFD015"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294955029,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#FFD015"
        fill="#FFD015"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#FFD015"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294955029,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#FFD015"
        fill="#FFD015"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#FFD015"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294955029,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight="16"
        bbWidth="16"
        color="#FFD015"
        fill="#FFD015"
        focusable={false}
        height={16}
        meetOrSlice={0}
        minX={0}
        minY={0}
        size={16}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            [
              undefined,
              undefined,
              {},
            ],
            {
              "flex": 0,
              "height": 16,
              "width": 16,
            },
          ]
        }
        tintColor="#FFD015"
        vbHeight={256}
        vbWidth={256}
        width={16}
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294955029,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGRect
            fill={null}
            height="256"
            propList={
              [
                "fill",
              ]
            }
            width="256"
            x="0"
            y="0"
          />
          <RNSVGPath
            d="M239.2,97.4A16.4,16.4,0,0,0,224.6,86l-59.4-4.1-22-55.5A16.4,16.4,0,0,0,128,16h0a16.4,16.4,0,0,0-15.2,10.4L90.4,82.2,31.4,86A16.5,16.5,0,0,0,16.8,97.4,16.8,16.8,0,0,0,22,115.5l45.4,38.4L53.9,207a18.5,18.5,0,0,0,7,19.6,18,18,0,0,0,20.1.6l46.9-29.7h.2l50.5,31.9a16.1,16.1,0,0,0,8.7,2.6,16.5,16.5,0,0,0,15.8-20.8l-14.3-58.1L234,115.5A16.8,16.8,0,0,0,239.2,97.4Z"
          />
        </RNSVGGroup>
      </RNSVGSvgView>
      <Text
        style={
          [
            {
              "marginLeft": 8,
            },
            {
              "color": "#18273C",
            },
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
          ]
        }
      >
        5
      </Text>
    </View>
    <View>
      <Text
        numberOfLines={2}
        style={
          [
            {
              "fontSize": 12,
              "lineHeight": 16,
            },
            {
              "color": "#6C757D",
            },
          ]
        }
      >
        fake-description
      </Text>
    </View>
  </View>
</View>
`;
